### PR TITLE
Set explicitly environment vars and other minor changes

### PR DESCRIPTION
**Pull request by alambike**
Set explicitly HOME environment var because in sudo executions, fails back to user home instead of root
Change gpg keyserver to pgp.mit.edu because is more reliable
Add 'not_if' guard to avoid download key if it was already in ring

### CODE REVIEW

File: recipes/system_install.rb [hunk 1]
```diff
@@ -30,10 +30,13 @@
   g.run_action(:create)
 end
 
+
 execute 'Adding gpg key' do
-  command "`which gpg2 || which gpg` --keyserver hkp://keys.gnupg.net --recv-keys #{node['rvm']['gpg_key']}"
+  environment ({"HOME" => "/root", "USER" => "root"})
+  command "`which gpg2 || which gpg` --keyserver hkp://pgp.mit.edu --recv-keys #{node['rvm']['gpg_key']}"
```
Comment (mpapis) [inline recipes/system_install.rb]: why the change of server? if one fails maybe allow configuring the server?

**Comment by alambike** [comment]
@mpapis Because seems more reliable (https://futureboy.us/pgp.html). Yesterday I have a lot of trouble with keys.gnupg.net. 
Anyway set it as a node attribute seems ok to me.

**Comment by mpapis** [comment]
I do not see any mention that `keys.gnupg.net` is not reliable in the mentioned doc, can you set `keys.gnupg.net` as default and update docs/readme about the new available key + description how to use the mit server?

**Comment by alambike** [comment]
Ok, I'll do it shortly when have a momment. 
Anyway I insists that keys.gnupg.net has some issues lastly ( see [here](https://github.com/phlipper/chef-percona/issues/235) or [here]( http://support.gpgtools.org/discussions/beta-feedback/59-problems-with-the-beta) for example, so other keyservers like mit or sks could be more adecuated to avoid issues.

**Comment by mpapis** [comment]
did you report the problem to `keys.gnupg.net`? it's one of the things that annoys me most - complaining somewhere else instead of "the failing thing" issue tracker - this is why I need to monitor twitter and stackoverflow for RVM bugs.

**Comment by alambike** [comment]
You're right, `keys.gnupg.net` is a cname to `pool.sks-keyservers.net`, so seems the more indicated to be the default. Apologizes for the discussion.

I would like to send the issue if I knew where and I could reproduce it. In that moment I thought it was a problem with the chef recipe or something mine, in fact was so because the HOME env var, but mixed with problems connecting to keyservers...  a mess that changing the keyserver solved part of it.
 
By the way, twitter and stackoverflow are good places to spend time ;-)

**Comment by mpapis** [comment]
you could try http://www.mail-archive.com/gpgtools-users@lists.gpgtools.org/

**Comment by mpapis** [comment]
@martinisoft this looks good now :+1:

**Comment by damacus** [comment]
## Closing due to inactivity. 

If this is still an issue please reopen or open another PR. 

Alternatively drop by the #sous-chefs channel on the [Chef Community Slack](http://community-slack.chef.io/) and we'll be happy to help! 

Thanks,
Sous-Chefs

**Pull request closed without merging**